org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/AnalyzeProjectJob.java

```diff
@@ -119,24 +132,85 @@ public void deleteUnmatched() throws CoreException {
   }
 
   @Override
-  protected IStatus run(SonarRunnerFacade runner, final IProgressMonitor monitor) {
-
-    // Configure
-    Properties properties = new Properties();
-    Collection<ProjectConfigurator> usedConfigurators = SonarLintAnalysisConfigurator.configureAnalysis(request, properties, extraProps, monitor);
+  protected IStatus run(SonarLintClientFacade sonarlint, final IProgressMonitor monitor) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "run" is 23 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)